vformer/encoder/convvt.py

```diff
@@ -0,0 +1,203 @@
+    depth: int
+        Number of CVT Attention blocks in each stage, default is 1
+    num_heads: int
+        Number of heads in attention, default is 6
+    mlp_ratio: float
+        Feature dimension expansion ratio in MLP, default is 4.0
+    p_dropout: float
+        Probability of dropout in MLP, default is 0.0
+    attn_dropout: float
+        Probability of dropout in attention, default is 0.0
+    drop_path_rate: float
+        Probability for droppath, default is 0.0
+    with_cls_token: bool
+        Whether to include classification token, default is False
+    kernel_size: int
+        Size of kernel, default is 3
+    padding_q: int
+        Size of padding in q, default is 1
+    padding_kv: int
+        Size of padding in kv, default is 2
+    stride_kv: int
+        Stride in kv, default is 2
+    stride_q: int
+        Stride in q, default is 1
+    init: str ('trunc_norm' or 'xavier')
+        Initialization method, default is 'trunc_norm'
+    """
+
+    def __init__(
+        self,
+        patch_size=7,
+        patch_stride=4,
+        patch_padding=0,
+        in_channels=3,
+        embedding_dim=64,
+        depth=1,
+        p_dropout=0.0,
+        drop_path_rate=0.0,
+        with_cls_token=False,
+        init="trunc_norm",
+        **kwargs
+    ):
+        super().__init__()
+
+        self.patch_embed = ConvEmbedding(
+            patch_size=patch_size,
+            in_channels=in_channels,
+            embedding_dim=embedding_dim,
+            stride=patch_stride,
+            padding=patch_padding,
+        )
+
+        if with_cls_token:
+            self.cls_token = nn.Parameter(torch.zeros(1, 1, embedding_dim))
+        else:
+            self.cls_token = None
+
+        self.pos_drop = nn.Dropout(p=p_dropout)
+        dpr = [
+            x.item() for x in torch.linspace(0, drop_path_rate, depth)
+        ]  # stochastic depth decay rule
+
+        blocks = []
+        for j in range(depth):
+            blocks.append(
+                ConvVTBlock(
+                    dim_in=embedding_dim,
+                    dim_out=embedding_dim,
+                    p_dropout=p_dropout,
+                    with_cls_token=with_cls_token,
+                    drop_path=dpr[j],
+                    **kwargs
+                )
+            )
+        self.blocks = nn.ModuleList(blocks)
+
+        if self.cls_token is not None:
+            trunc_normal_(self.cls_token, std=0.02)
+
+        if init == "xavier":
+            self.apply(self._init_weights_xavier)
+        elif init == "trunc_norm":
+            self.apply(self._init_weights_trunc_normal)
+        else:
+            raise ValueError("Init method {} not found".format(init))
+
+    def _init_weights_trunc_normal(self, m):
+        if isinstance(m, nn.Linear):
+            trunc_normal_(m.weight, std=0.02)
+        elif isinstance(m, (nn.LayerNorm, nn.BatchNorm2d)):
+            nn.init.constant_(m.bias, 0)
+            nn.init.constant_(m.weight, 1.0)
```

Comment:
Thanks for the PR!
Here we can use `trunc_normal_` function of the timm library as timm is one of the dependencies; 

or else we can add this as utility function in utils module.

Also we are using only DropPath class of timm other than `trunc_normal_`; so maybe we can remove timm dependency by implementing `DropPath` class in the library.


@NeelayS

Reply:
I think we would be discrediting the `timm` if we just copied over their code. I know it adds an additional dependency, but I think we should use it from their package.

Reply:
I already used the `trunc_normal_` from `timm`, I think the authors wanted to initialize it differently on different layers, hence the function.